driver/mobile/keyboard.go

```diff
@@ -0,0 +1,26 @@
+package mobile
+
+import (
+	"fyne.io/fyne"
+)
+
+// Keyboard represents a standard fyne keyboard
+type Keyboard int32
+
+const (
+	// DefaultKeyboard is the keyboard with default input style and "Done" return key
+	DefaultKeyboard Keyboard = iota
+	// MultiLineKeyboard is the keyboard with default input style and "return" return key
+	MultiLineKeyboard
+	// NumberKeyboard is the keyboard with number input style and "Done" return key
+	NumberKeyboard
+	// WebKeyboard is the keyboard with web input style and "Done" return key
+	WebKeyboard
+)
+
+// Keyboardable describes any CanvasObject that needs a keyboard
+type Keyboardable interface {
+	fyne.Focusable
+
+	Keyboard() Keyboard
```

Comment:
The duplication of naming here implies we need to think better `Keyboard` is a shorthand for `GetKeyboard` but then the return type doesn't help understand the context, what is a `Keyboard` - a layout of buttons?

Reply:
I agree, it doesn't look good.
I also had thoughts about more flexible api, something like that:
```go
const (
	// DefaultKeyboard is the keyboard with default input style and "Done" return key
	DefaultKeyboard Keyboard = 1 << iota
	// MultiLineKeyboard is the keyboard modifier which replaces return key with "return"
	MultiLineKeyboard
	// NumberKeyboard is the keyboard modifier which replaces input style to numbers only
	NumberKeyboard
	// PhoneKeyboard is the keyboard modifier which replaces input style to phone keys only
	PhoneKeyboard
)
```
And someone can use it like this: `ShowVirtualKeyboard(NumberKeyboard | MultiLineKeyboard)` or `ShowVirtualKeyboard(DefaultKeyboard)`. It is more complicated to use and implement but has its advantages.

Reply:
I don't think that building up flags like this makese sense. I can't think what a "multiline number" keyboard is as the number keys don't have a carriage return.
Perhaps you can expand on what the advantages are that you hint at?